remotespark/livyclientlib/livysession.py

```diff
@@ -106,11 +107,38 @@ def create_sql_context(self):
         else:
             raise FailedToCreateSqlContextException("Failed to create the SqlContext.\nError: '{}'".format(out))
 
+    def get_app_id(self):
+        if self._app_id is None:
+            session = self._http_client.get_session(self.id)
+            self._app_id = session['appId']
```

Comment:
Depends on Livy version obviously.  Not a problem for HDInsight but a problem if a person installs this repo and happens to be running against an old Livy version.  Don't know what the implications of this are, we haven't really discussed our approach to Livy versioning.